File: core/src/main/java/com/github/nomad145/purepazaak/core/entity/PlusCard.java
package com.github.nomad145.purepazaak.core.entity;

import com.github.nomad145.purepazaak.core.card.CardInterface;

public class PlusCard implements CardInterface {
  protected int value;

  public PlusCard(int value) {
    this.value = value;
  }

  public int getValue() {
    return value;
  }

  public String getImagePath() {
    return "cards/plus.png";
  }
}

File: core/src/main/kotlin/com/github/nomad145/purepazaak/core/card/PlusCard.kt
package com.github.nomad145.purepazaak.core.card;

import java.lang.IllegalArgumentException;

open class PlusCard(override val value: Int) : CardInterface {
  init {
    if (value < 0) {
      throw IllegalArgumentException("The card value must be positive");
    }
  }
}

